vmware-tanzu/tanzu-framework
# change addons-manager package name to v2

### What this PR does / why we need it
change addons-manager package name to addons-manager-v2

### Which issue(s) this PR fixes


Fixes #1720 

### Describe testing done for PR

1. Upgrade golang version to 1.17+
2. run make build-install-cli-all to create local cli builds
3. set OCI_REGISTRY for local testing, which is to set registry to hold built images
4. make docker-all to build the images, publish the images and update kbld-config.yml in the repo
5. make local-registry to setup local resgistry
6. make management-package-bundles to build tar bundles for packages
7. make push-management-package-bundles to push management packages to registry
8. make package-repo-bundle to build tar bundles for package repo
9. make push-management-package-repo-bundle to push management package repository bundle to registry
10. set up a test cluster and install the cli built in step 2
11. use tanzu cli to add management package repository bundle into the test cluster by using "tanzu package repository add test-repo --url xxxxxx"
12. verify the reconcile is succeed by using "tanzu package repository list -A"
13. verify addons-manager package is inside the package bundle by using "tanzu package available list -A"



### Release note

```release-note
Add addons manager package in management package test result

tanzu package repository list -A
NAME              REPOSITORY                                                     TAG                                                                      STATUS               DETAILS  NAMESPACE
tanzu-test        index.docker.io/pksoctopus/management                          sha256:466d09f3e3bd09397ac4c0830111ea8849b9b113a031815b6a5e8012fbde5280  Reconcile succeeded           default

tanzu package available list -A
NAME                                                DISPLAY-NAME                       SHORT-DESCRIPTION                                                                                                            LATEST-VERSION                NAMESPACE
addons-manager.tanzu.vmware.com                     tanzu-addons-manager               This package provides TKG addons lifecycle management capabilities.                                                          0.16.0-dev                    default
core-management-plugins.tanzu.vmware.com            core-management-plugins            CLIPlugin management                                                                                                         0.16.0-dev                    default
featuregates.tanzu.vmware.com                       featuregates                       Feature management                                                                                                           0.16.0-dev                    default
```

### PR Checklist



      


### Additional information

#### Special notes for your reviewer
